### PR TITLE
fix: Add error property to CartAddEntryFailEvent

### PR DESCRIPTION
**Pull request by marlass**
- added `error` property to `CartAddEntryFailEvent`
- changed `HttpErrorModel` from interface to class (to be able to use `instanceof` operator for type safety)
- adjusted `normalizeHttpError` method to return instance of the `HttpErrorModel` instead of plain object
- improved type safety in `createQuickOrderResultEvent` method
- leveraged the `HttpErrorModel` class to improve type correctness in `ResetPasswordComponentService`

### CODE REVIEW

File: projects/core/src/util/normalize-http-error.spec.ts [hunk 1]
```diff
@@ -22,12 +22,11 @@ describe('normalizeHttpError', () => {
...
-        message: mockError.message,
-        status: mockError.status,
-        statusText: mockError.statusText,
-        url: mockError.url,
-      } as HttpErrorModel);
+      expect(result?.message).toEqual(mockError.message);
+      expect(result?.status).toEqual(mockError.status);
+      expect(result?.statusText).toEqual(mockError.statusText);
+      expect(result?.url).toEqual(mockError.url);
+      expect(result instanceof HttpErrorModel).toBeTruthy();
```
Comment (znikola) [inline projects/core/src/util/normalize-http-error.spec.ts]: What do you think about using this assertion style:

```ts
   expect(result).toEqual(
        jasmine.objectContaining({
          message: mockError.message,
          status: mockError.status,
          statusText: mockError.statusText,
          url: mockError.url,
        })
      );
```

File: feature-libs/cart/quick-order/core/services/quick-order.service.ts [hunk 2]
```diff
@@ -195,15 +196,21 @@ export class QuickOrderService implements QuickOrderFacade {
     let evt: QuickOrderAddEntryEvent = {
```
Comment (znikola) [inline feature-libs/cart/quick-order/core/services/quick-order.service.ts]: can we change it to `const`? 🤔 
I see that we only mutate the variable, and re-assign it.

File: feature-libs/cart/quick-order/core/services/quick-order.service.ts [hunk 3]
```diff
@@ -195,15 +196,21 @@ export class QuickOrderService implements QuickOrderFacade {
...
+    }
+    if ('quantityAdded' in cartEvent) {
+      evt.quantityAdded = cartEvent.quantityAdded;
+    }
+    if ('error' in cartEvent && cartEvent.error instanceof HttpErrorModel) {
+      evt.error = cartEvent.error;
+    }
+
     if (evt.error?.details?.length) {
       let isOutOfStock = evt.error?.details.some(
```
Comment (znikola) [inline feature-libs/cart/quick-order/core/services/quick-order.service.ts]: The same as above - can we change it to `const`?

File: feature-libs/user/profile/components/reset-password/reset-password-component.service.ts [hunk 4]
```diff
@@ -81,16 +80,18 @@ export class ResetPasswordComponentService {
     this.redirect();
   }
 
-  protected onError(error: Error): void {
+  protected onError(error: unknown): void {
```
Comment (znikola) [inline feature-libs/user/profile/components/reset-password/reset-password-component.service.ts:83]: Is this considered as a breaking change?

**Review by znikola — CHANGES_REQUESTED** [review]

File: feature-libs/user/profile/components/reset-password/reset-password-component.service.ts (hunk 4 above)
Comment (marlass) [reply feature-libs/user/profile/components/reset-password/reset-password-component.service.ts:83]: I would consider it a bugfix. It wasn't correct from the beggining. normalized HTTP error errors weren't of this type.

**Review by znikola — APPROVED** [review]

**Pull request merged**